projects/packages/videopress/src/client/lib/get-media-token/index.native.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+/**
+ * Internal dependencies
+ */
+import { MEDIA_TOKEN_SCOPES } from './types';
+/**
+ * Types
+ */
+import type { MediaTokenScopeProps, MediaTokenProps, GetMediaTokenArgsProps } from './types';
+
+type Response = {
+	metadata_token: string;
+	blog_id: string;
+};
+
+/**
+ * Request media token data hiting WPCOM API.
+ *
+ * @param {MediaTokenScopeProps} scope  - The scope of the token to request.
+ * @param {GetMediaTokenArgsProps} args - function arguments.
+ * @returns {MediaTokenProps}             Media token data.
+ */
+const requestMediaToken = function (
+	scope: MediaTokenScopeProps,
+	args: GetMediaTokenArgsProps = {}
+): Promise< MediaTokenProps > {
+	const { guid } = args;
+	return new Promise( function ( resolve, reject ) {
+		if ( ! MEDIA_TOKEN_SCOPES.includes( scope ) ) {
+			return reject( 'Invalid scope' );
+		}
+
+		const fetchParams: { path: string; body: object } = { path: '', body: {} };
+
+		switch ( scope ) {
+			case 'upload':
```

Comment:
Well observed @SiobhyB, what you shared matches exactly the mental process I followed when writing this part 🙂.

I started implementing the native version by reusing the code from the web, my goal was to replicate as much of the logic as possible to avoid diverging the code. However, it ended up being quite different due to the lack of support for `upload` and `upload-jtw` scope items, but still, I wanted to keep it look similar.

Not sure if in the future we'll support those scope items, in any case, rejecting them explicitly here I think it would serve as a reminder that we don't support them in native 😅.

Reply:
@fluiddot, that makes a lot of sense and I really appreciate you expanding on your thought process here, too, thank you!